--- mo/extensions/load/onnx/loader.py	2021-11-16 19:44:00.000000000 +0700
+++ venv39/lib/python3.9/site-packages/mo/extensions/load/onnx/loader.py	2022-01-27 11:51:29.399203519 +0700
@@ -54,9 +54,12 @@
         graph.graph['fw'] = 'onnx'
         graph.graph['feature_dim'] = 1
         if hasattr(model_proto, 'opset_import'):
-            graph.graph['fw_opset_version'] = model_proto.opset_import[0].version   # pylint: disable=no-member
-        else:
-            graph.graph['fw_opset_version'] = None
+            opset_imports = {
+                one_opset_import.domain: one_opset_import.version
+                for one_opset_import in model_proto.opset_import  # pylint: disable=no-member
+            }
+            # load opset version with default domain
+            graph.graph['fw_opset_version'] = opset_imports.get('', None)
 
         graph.check_empty_graph('protobuf2nx. It may happen due to problems with loaded model')
         extract_node_attrs(graph, lambda node: onnx_op_extractor(node, check_for_duplicates(onnx_op_extractors)))
