diff --git a/ModelImporter.cpp b/ModelImporter.cpp
index 93c5614..393d187 100644
--- a/ModelImporter.cpp
+++ b/ModelImporter.cpp
@@ -509,10 +509,10 @@ bool ModelImporter::supportsOperator(const char* op_name) const
 
     if (is("NonMaxSuppression"))
     {
-        return false;
+        return true;
     }
     if (is("EfficientNMS_TRT") || is("PyramidROIAlign_TRT") || is("MultilevelCropAndResize_TRT")
-        || is("DisentangledAttention_TRT"))
+        || is("DisentangledAttention_TRT") || is("RoiAlign"))
     {
         return true;
     }
