diff --git a/ModelImporter.cpp b/ModelImporter.cpp
index 7f55a51..c9c20ad 100644
--- a/ModelImporter.cpp
+++ b/ModelImporter.cpp
@@ -499,9 +499,9 @@ bool ModelImporter::supportsOperator(const char* op_name) const
 
     if (is("NonMaxSuppression"))
     {
-        return false;
+        return true;
     }
-    if (is("EfficientNMS_TRT") || is("PyramidROIAlign_TRT") || is("MultilevelCropAndResize_TRT"))
+    if (is("EfficientNMS_TRT") || is("PyramidROIAlign_TRT") || is("MultilevelCropAndResize_TRT") || is("RoiAlign"))
     {
         return true;
     }
