diff --git a/ModelImporter.cpp b/ModelImporter.cpp
index 720f4ce..5cd61db 100644
--- a/ModelImporter.cpp
+++ b/ModelImporter.cpp
@@ -467,7 +467,7 @@ bool ModelImporter::supportsOperator(char const* op_name) const
     // Disable nodes that rely on DDS as ONNXRuntime does not support it at the moment
     if (is("NonMaxSuppression") || is("NonZero") || is("RoiAlign"))
     {
-        return false;
+        return true;
     }
     return _op_importers.count(op_name);
 }
