diff --git a/onnxruntime/python/tools/quantization/CalTableFlatBuffers/TrtTable.py b/onnxruntime/python/tools/quantization/CalTableFlatBuffers/TrtTable.py
index 5c70301e8..dcb837256 100644
--- a/onnxruntime/python/tools/quantization/CalTableFlatBuffers/TrtTable.py
+++ b/onnxruntime/python/tools/quantization/CalTableFlatBuffers/TrtTable.py
@@ -28,7 +28,7 @@ class TrtTable(object):
             x = self._tab.Vector(o)
             x += flatbuffers.number_types.UOffsetTFlags.py_type(j) * 4
             x = self._tab.Indirect(x)
-            from CalTableFlatBuffers.KeyValue import KeyValue
+            from onnxruntime.quantization.CalTableFlatBuffers.KeyValue import KeyValue
             obj = KeyValue()
             obj.Init(self._tab.Bytes, x)
             return obj
